scripts/env.sh

```diff
@@ -0,0 +1,25 @@
+
+DIR=$(pwd)
+
+# for mongodbtoolchain python3
+export PATH=/opt/mongodbtoolchain/v3/bin:$PATH
+
+# Check if not called with -g or --run-global flag
+if [[ "$*" != *-g* ]]; then
+    if [[ ! -e "$SCRIPTS_DIR/venv/.setup-complete" ]]; then
+        echo "Installing virtual environment."
+        echo "To run without a virtual environment, use the -g or --run-global option."
+        echo "creating new env with python: $(command -v python3)"
+        pushd "$SCRIPTS_DIR" >/dev/null
+            python3 -m venv venv
+        popd >/dev/null
+        touch "$SCRIPTS_DIR/venv/.setup-complete"
+    fi
+
+    set +u
+        source "${SCRIPTS_DIR}/venv/bin/activate"
+        pushd "$SCRIPTS_DIR/../src/python" >/dev/null
+            python3 setup.py --quiet develop
+        popd >/dev/null
```

Comment:
this has the down-side of calling setup every time the `./scripts/genny` call is made. I think we can do his inside of the "if not already setup" block right before touching the setup-complete file.

Reply:
My only fear here was I can just see someone getting hurt by an out of date environment after someone changes setup.py, and wasting quite a bit of time trying to investigate.

 I'm guessing this was also a risk beforehand as well? I'm gonna defer to you here.

Reply:
Ya good point, but I think that's fine. If somebody's explicitly modifying setup.py I don't think it's unreasonable to require them to run it directly (and they're likely going to have to understand this shell-python-shell triple-indirection anyway 🙄😸)